### DISCUSSION TITLE
Setup with RPi Pico hardware

### DISCUSSION THREAD

**Discussion by tjboldt** [tjboldt/Apple2-IO-RPi#162] [General]
These instructions are for using the Pico code from Oliver Schmidt and hardware from Ralle Palaveev

1. Flash [Apple2-IO-RPi.uf2](https://github.com/oliverschmidt/apple2-io-rpi/releases/latest/download/Apple2-IO-RPi.uf2) to the Raspberry Pi Pico.

2. Execute steps __5.)__ - __8.)__ and __10.)__ on [Setup new card from scratch](https://github.com/tjboldt/Apple2-IO-RPi/discussions/63).

3. Execute `wget --no-cache -O - https://raw.githubusercontent.com/oliverschmidt/apple2-io-rpi/main/RaspberryPi/setup.sh | bash`

**Comment by DanBrownleeOps** [comment]
Just got my Pico and other personalities seem fine but for the Apple2-IO-RPI, I get waiting for RPi FW:8010. What am I missing? I saw no errors in the steps above.

**Reply by DanBrownleeOps** [reply]
Figured out I needed "-cdc" on apple2driver. Boots ProDOS now but I get a hang on -shell with "invalid escape sequence" message on Pi side.

**Reply by tjboldt** [reply]
I am updating the setup and documentation in another branch already and hope to merge that in this week

**Reply by tjboldt** [reply]
As for the shell, you need to assemble a version for the Pico hardware…

In the Apple2 directory, run:

./assemble.sh 1

The 1 specifies Pico hardware. 0 is the default and is for the classic hardware

**Reply by DanBrownleeOps** [reply]
Magic! Thank you! Working now.
